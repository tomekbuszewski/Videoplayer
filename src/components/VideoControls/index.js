/**
 * @flow
 * @author tomekbuszewski
 * @since 2018-11-27
 */

import * as React from 'react';
import inject from 'react-jss';
import c from 'classnames';

import GenericButton from '@src/components/GenericButton';
import TogglePlay from '@src/components/TogglePlayButton';
import Timeline from '@src/components/Timeline';

import AudioOnIcon from '@assets/audio-on.svg';
import AudioOffIcon from '@assets/audio-off.svg';

import styles from './index.styles';

type Props = {
  classes: Object,
  duration: number,
  highlights?: Object[],
  muted: boolean,
  paused: boolean,
  position: number,

  togglePlay: Function,
  toggleMute: Function,
  setPlayback: Function,
};

const VideoControls = (props: Props) => {
  const {
    classes,
    duration,
    highlights,
    muted,
    paused,
    position,

    togglePlay,
    toggleMute,
    setPlayback,
  } = props;

  return (
    <div className={c(classes.ControlsWrapper, paused && classes.ControlsWrapperActive)}>
      <TogglePlay onClick={togglePlay} paused={paused} />
      <div className={classes.Controls}>
        <Timeline
          duration={duration}
          highlights={highlights}
          onClick={setPlayback}
          position={position}
        />
        <GenericButton
          active={muted}
          activeState={<AudioOffIcon />}
          inactiveState={<AudioOnIcon />}
          onClick={toggleMute}
        />
      </div>
    </div>
  );
};

VideoControls.defaultProps = {
  highlights: null,
};

export default inject(styles)(VideoControls);
