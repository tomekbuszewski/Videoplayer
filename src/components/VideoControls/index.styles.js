import { TRANSITION_SHORTHAND } from '@config/animations';
import { COLOR_BLACK } from '@config/colors';
import { GET_STEP, STEP_MEDIUM } from '@config/visuals';

export default {
  ControlsWrapper: {
    opacity: 0,
    pointerEvents: 'mouse',
    transition: TRANSITION_SHORTHAND('opacity'),

    '&:hover': {
      opacity: 1,
    },
  },

  ControlsWrapperActive: {
    opacity: 1,
    pointerEvents: 'all',
  },

  Controls: {
    position: 'absolute',
    bottom: 0,
    left: 0,
    width: '100%',
    background: `linear-gradient(to bottom, transparent 0%, ${COLOR_BLACK} 100%); `,
    display: 'flex',
    alignItems: 'center',

    padding: GET_STEP(STEP_MEDIUM),
  },
};
