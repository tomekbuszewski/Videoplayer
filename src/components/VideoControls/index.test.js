import * as React from 'react';
import renderer from 'react-test-renderer';

import { highlights } from '@src/containers/Video/index.stories';

import VideoControls from '.';

describe('VideoControls tests', () => {
  it('should match snapshot with highlights', () => {
    const component = renderer.create(<VideoControls
      duration={100}
      highlights={highlights}
      muted
      paused
      position={30}

      togglePlay={() => false}
      toggleMute={() => false}
      setPlayback={() => false}
    />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot without highlights', () => {
    const component = renderer.create(<VideoControls
      duration={100}
      muted
      paused
      position={30}

      togglePlay={() => false}
      toggleMute={() => false}
      setPlayback={() => false}
    />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
