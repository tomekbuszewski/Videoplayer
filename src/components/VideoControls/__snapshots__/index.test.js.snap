// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoControls tests should match snapshot with highlights 1`] = `
<div
  className="VideoControls-ControlsWrapper-0-1-1 VideoControls-ControlsWrapperActive-0-1-2"
>
  <button
    className="PlayPause-Button-0-1-4"
    onClick={[Function]}
    type="button"
  />
  <div
    className="VideoControls-Controls-0-1-3"
  >
    <div
      className="Timeline-Timeline-0-1-5"
    >
      <div
        className="Timeline-Track-0-1-6"
      />
      <div
        className="Timeline-Pointer-0-1-7"
        style={
          Object {
            "left": "30%",
          }
        }
      />
      <div
        className="Highlight-Highlight-0-1-8"
        style={
          Object {
            "left": "100%",
          }
        }
      >
        <button
          className="Highlight-Pointer-0-1-9"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          type="button"
        />
        <div
          className="Highlight-CaptionHolder-0-1-10"
        >
          <p>
            Lorem ipsum
          </p>
        </div>
      </div>
      <div
        className="Highlight-Highlight-0-1-8"
        style={
          Object {
            "left": "200%",
          }
        }
      >
        <button
          className="Highlight-Pointer-0-1-9"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          type="button"
        />
        <div
          className="Highlight-CaptionHolder-0-1-10"
        >
          <img
            alt=""
            src="https://www.fillmurray.com/300/200"
          />
          <p>
            Dolor sit
          </p>
        </div>
      </div>
    </div>
    <button
      className="GenericButton-GenericButton-0-1-12"
      onClick={[Function]}
      type="button"
    />
  </div>
</div>
`;

exports[`VideoControls tests should match snapshot without highlights 1`] = `
<div
  className="VideoControls-ControlsWrapper-0-1-1 VideoControls-ControlsWrapperActive-0-1-2"
>
  <button
    className="PlayPause-Button-0-1-4"
    onClick={[Function]}
    type="button"
  />
  <div
    className="VideoControls-Controls-0-1-3"
  >
    <div
      className="Timeline-Timeline-0-1-5"
    >
      <div
        className="Timeline-Track-0-1-6"
      />
      <div
        className="Timeline-Pointer-0-1-7"
        style={
          Object {
            "left": "30%",
          }
        }
      />
    </div>
    <button
      className="GenericButton-GenericButton-0-1-12"
      onClick={[Function]}
      type="button"
    />
  </div>
</div>
`;
